### PR TITLE
[yang] Add Bmc to Device Neighbor Metadata element type list

### PR DESCRIPTION
**Pull request by yaqiangz**
#### Why I did it
Bmc is a valid neighbor type in minigraph, however it was missing from the YANG model definition. Usually, the Bmc type device can be neighbor of BmcMgmtToRRouter. This PR is to introduce this type.

##### Work item tracking
- Microsoft ADO **(number only)**: 24869573

#### How I did it
Add `Bmc` to element type list.

#### How to verify it
UT passed.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by ganglyu — APPROVED** [review]

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/device_neighbor_metadata.json [hunk 1]
```diff
@@ -25,6 +25,32 @@
             }
         }
     },
+    "DEVICE_NEIGHBOR_METADATA_TYPE_INCORRECT_PATTERN_BMCS": {
+        "sonic-device_neighbor_metadata:sonic-device_neighbor_metadata": {
+            "sonic-device_neighbor_metadata:DEVICE_NEIGHBOR_METADATA": {
+                "DEVICE_NEIGHBOR_METADATA_LIST": [
+                    {
+                        "name": "Ethernet116",
+                        "hwsku": "Arista",
```
Comment (Blueve) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/device_neighbor_metadata.json:34]: DUMMY_BMC_SKU

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/device_neighbor_metadata.json [hunk 2]
```diff
@@ -25,6 +25,32 @@
...
+                        "type": "Bmcs"
```
Comment (Blueve) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/device_neighbor_metadata.json:35]: InvalidBmcType

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/device_neighbor_metadata.json (hunk 1 above)
Comment (yaqiangz) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/device_neighbor_metadata.json]: Fixed

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/device_neighbor_metadata.json (hunk 2 above)
Comment (yaqiangz) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/device_neighbor_metadata.json]: Fixed

**Comment by yaqiangz** [comment]
> Could you add unit test?

@qiluo-msft Added, could you please help to review/merge this PR?

**Review by Blueve — APPROVED** [review]

**Pull request merged**